app/src/main/java/com/kpiroom/bubble/source/api/impl/firebase/FirebaseApi.kt

```diff
@@ -11,18 +16,43 @@ class FirebaseApi : ApiInterface {
     })
 
     private val authUtil = FirebaseAuthUtil(FirebaseAuth.getInstance())
+    private val storageUtil = FirebaseStrorageUtil(FirebaseStorage.getInstance())
 
+    //Database
     override suspend fun getServerVersion(): String = dbUtil.read(FirebaseStructure.VERSION, String::class.java)
 
-    override suspend fun setServerVersion(version: String) = dbUtil.write(FirebaseStructure.VERSION, version)
+    override suspend fun setServerVersion(version: String): Unit = dbUtil.write(FirebaseStructure.VERSION, version)
 
-    override suspend fun signUp(email: String, password: String) = authUtil.signUp(email, password).also {
-        it?.let { id ->
-            dbUtil.write("${FirebaseStructure.USER(id)}", id)
+    override suspend fun usernameExists(username: String): Boolean = dbUtil.usernameExists(username)
+
+    override suspend fun getUserData(uuid: String?): User? = dbUtil.getUserData(uuid)
```

Comment:
Remove nullable String uuid. It's non Kotlin style and it's not applicable for this context